src/main/java/plugily/projects/villagedefense/kits/premium/WizardKit.java

```diff
@@ -93,117 +104,215 @@ public Material getMaterial() {
+            VersionUtils.sendParticles("SUSPENDED", null, en.getLocation(), 1, 0, 0, 0);
+            totalDamage += damage;
+          }
+          player.setHealth(Math.min(player.getHealth() + (totalDamage * 0.08), VersionUtils.getMaxHealth(player)));
+        }
+        if (damageTick % 10 == 0) {
+          VersionUtils.sendActionBar(player, messages.get(messageIndex)
+              .replace("%number%", String.valueOf(user.getCooldown("wizard_bloodlust_running"))));
+          messageIndex++;
+          if (messageIndex > messages.size() - 1) {
+            messageIndex = 0;
+          }
+        }
+        if (damageTick >= 20 * 15 || !getPlugin().getArenaRegistry().isInArena(user.getPlayer()) || user.isSpectator()) {
+          //reset action bar
+          VersionUtils.sendActionBar(player, "");
           cancel();
+          return;
         }
+        damageTick++;
       }
-    }.runTaskTimer(getPlugin(), 0, 2);
+    }.runTaskTimer(getPlugin(), 0, 1);
   }
 
-  private void applyMagicAttack(Player player) {
+  private void applyMagicAttack(User user) {
+    XParticle.drawLine(user.getPlayer(), 40, 1, ParticleDisplay.of(XParticle.getParticle("FLAME")));
+    Arena arena = (Arena) user.getArena();
+    final int finalPierce = getDefaultPierce(arena);
     new BukkitRunnable() {
-      double positionModifier = 0;
-      final Location loc = player.getLocation();
+      final Location loc = user.getPlayer().getLocation();
       final Vector direction = loc.getDirection().normalize();
+      double positionModifier = 0;
+      int pierce = finalPierce;
 
       @Override
       public void run() {
-        positionModifier += 0.5;
+        positionModifier += 3.5;
         double x = direction.getX() * positionModifier,
             y = direction.getY() * positionModifier + 1.5,
             z = direction.getZ() * positionModifier;
         loc.add(x, y, z);
-        VersionUtils.sendParticles("TOWN_AURA", null, loc, 5, 0, 0, 0);
-        for(Entity en : loc.getChunk().getEntities()) {
-          if(!(CreatureUtils.isEnemy(en)) || en.getLocation().distance(loc) >= 1.5 || en.equals(player)) {
+        for (Entity en : loc.getChunk().getEntities()) {
+          if (!CreatureUtils.isEnemy(en) || en.getLocation().distance(loc) >= 1.8 || en.equals(user.getPlayer()) || pierce <= 0) {
             continue;
           }
-          ((LivingEntity) en).damage(6.0, player);
-          VersionUtils.sendParticles("FIREWORKS_SPARK", null, en.getLocation(), 2, 0.5, 0.5, 0.5);
+          LivingEntity entity = (LivingEntity) en;
+          //wand damage: 30/32/35 scaling % of entity's max health, we use direct health set to override armors etc.
+          entity.damage(0, user.getPlayer());
+          double maxHealthPercent = getWandPercentageDamage(arena);
+          entity.setHealth(Math.max(0, entity.getHealth() - (entity.getAttribute(Attribute.GENERIC_MAX_HEALTH).getValue() / 100.0) * maxHealthPercent));
```

Comment:
Use VersionUtil getMaxHealth Attributes will cause issues on direct call for legacy versions